src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -77,37 +76,44 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
       $query->condition($bundle_key, $bundles, 'IN');
     }
 
-    $user_groups = $this->getUserGroups();
-    if (!$user_groups) {
+    $user = User::load(\Drupal::currentUser()->id());
+    if ($user->hasPermission('administer group')) {
+      // User can see all the groups.
       return $query;
     }
 
+    // User is a non-site wide group admin.
     $identifier_key = $definition->getKey('id');
 
+    /** @var \Drupal\og\OgAccessInterface $og_access */
+    $og_access = \Drupal::service('og.access');
+
+    if (empty($this->configuration['entity'])) {
+      $entity_type_id = $this->configuration['entity_type_id'];
+      $bundle = $this->configuration['bundle'];
+    }
+    else {
+      /** @var \Drupal\Core\Entity\EntityInterface $entity */
+      $entity = $this->configuration['entity'];
+      $entity_type_id = $entity->getEntityTypeId();
+      $bundle = $entity->bundle();
+    }
+
     $ids = [];
-    if (!empty($this->configuration['handler_settings']['field_mode']) && $this->configuration['handler_settings']['field_mode'] == 'admin') {
-      // Don't include the groups, the user doesn't have create permission.
-      foreach ($user_groups as $delta => $group) {
+    foreach ($this->getUserGroups() as $group) {
+      // Check user has "create" permission on this entity.
+      if ($og_access->userAccess($group, "create $entity_type_id $bundle", $user)->isAllowed()) {
```

Comment:
@pfrenssen If I remember correctly `\Drupal\og\EventSubscriber\OgEventSubscriber::provideDefaultNodePermissions` overrides the permissions for nodes so they will be in the format of `create foo content`. However that will mean we'll have to think about it everywhere.

So maybe we should remove it, and also for node follow the `create [entity-type] [bundle]` pattern. What do you think?

Reply:
@amitaibu,

This seems to be a critical issue atm. Can we revert to the format `create foo content` here and refactor to `create [entity-type] [bundle]` in a follow up issue?

Reply:
From my perspective it seems fine, but need to get approval from @pfrenssen  based on above comment - https://github.com/Gizra/og/pull/181#discussion_r79059916

Reply:
This is already solved for group content but apparently not for groups.

In order to avoid the problems that arise from using this kind of string based permissions we have provided a flexible permission system that tracks operations and group content entity types. The idea behind this is that in order to discover the right permission to use we should not be messing with constructing strings but discover `PermissionInterface` objects that map to entity types and operations.

You can check `GroupContentOperationPermission` to see how we solved this for group content. If you want to know whether a user has permission to access the "create" operation on a group content entity you can call `OgAccess::userAccessGroupContentEntityOperation()` and pass the 'create' operation for this content type.

Reply:
@pfrenssen, thanks for the update. I assume this would work:
```suggestion
      if ($this->ogAccess->userAccessGroupContentEntityOperation('create', $group, $entity, $this->currentUser)->isAllowed()) {
```